From c844abae2756d0e57a5358e4c6764cc79df43bfe Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 7 Jul 2022 15:25:17 +0200
Subject: [PATCH 0274/1141] drm/stm: ltdc: set max pad to 150MHz

Need a pad max frequency of 150MHz for 3 layers.

Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Change-Id: I4f25597e44f037b7a0c8f3a83993b2b35e41f1e7
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/262861
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Raphael GALLAIS-POU <raphael.gallais-pou@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Raphael GALLAIS-POU <raphael.gallais-pou@foss.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -1917,7 +1917,11 @@ static int ltdc_get_caps(struct drm_devi
 		ldev->caps.pix_fmt_nb = ARRAY_SIZE(ltdc_drm_fmt_a2);
 		ldev->caps.pix_fmt_flex = true;
 		ldev->caps.non_alpha_only_l1 = false;
-		ldev->caps.pad_max_freq_hz = 90000000;
+		/* Need a pad max frequency of 150MHz for 3 layers */
+		if (lcr == 3)
+			ldev->caps.pad_max_freq_hz = 150000000;
+		else
+			ldev->caps.pad_max_freq_hz = 90000000;
 		ldev->caps.nb_irq = 2;
 		ldev->caps.ycbcr_input = true;
 		ldev->caps.ycbcr_output = true;
